sway-core/src/semantic_analysis/type_check_context.rs

```diff
@@ -1201,6 +1201,7 @@ impl<'a> TypeCheckContext<'a> {
             }
 
             if !maybe_method_decl_refs.is_empty() {
+                #[allow(clippy::mutable_key_type)]
```

Comment:
Maybe we shouldn't enforce this clippy lint since we need to disable it in quite a few places...